authorize_helper.go

```diff
@@ -41,11 +41,17 @@ import (
 func GetRedirectURIFromRequestValues(values url.Values) (string, error) {
 	// rfc6749 3.1.   Authorization Endpoint
 	// The endpoint URI MAY include an "application/x-www-form-urlencoded" formatted (per Appendix B) query component
-	redirectURI, err := url.QueryUnescape(values.Get("redirect_uri"))
+	redirectURI, err := url.Parse(values.Get("redirect_uri"))
```

Comment:
I think this is correct. We just want to validate the `redirect_uri` value.

Reply:
I'm also wondering if `values.Get` isn't already unescaped?

Reply:
Yes. That is the root of the issue.